app/Http/Controllers/CourseController.php

```diff
@@ -59,33 +65,31 @@ public function create()
 	public function store(Request $request)
 	{
 		$attributes = request()->validate([
-			'courseName' => self::REQUIRED,
+			'courseName' => 'required',
 			'courseType' => 'required|integer|min:1|max:2',
 			'teamSize' => 'required|integer|min:1',
-			self::TEACHERS => 'required|array|min:1',
+			'teachers' => 'required|array|min:1',
 			'teachers.*' => [
 				'integer',
-				Rule::in(User::where('role', config(self::ROLES)[self::TEACHER])->get()->pluck('id')),
+				Rule::in(User::where('role', config(self::ROLES)['teacher'])->get()->pluck('id')),
 			],
-			'courseSemester' => self::REQUIRED,
+			'courseSemester' => 'required',
 			'courseSchedule' => 'required|array|min:1',
-			'courseHours' => self::REQUIRED,
-			self::ASSOCIATED => 'nullable|array|min:1',
+			'courseHours' => 'required',
+			'associatedCourses' => 'nullable|array|min:1',
 		]);
 
 		$course = $this->createCourse($attributes);
-		if (!isset($course)) {
-			abort(500);
-		}
 
-		$courseKey = $course->course_key;
-
-		$course->teachers()->attach($attributes[self::TEACHERS]);
-		if (isset($attributes[self::ASSOCIATED])) {
-			$course->suppliers()->attach($attributes[self::ASSOCIATED]);
+		$course->teachers()->attach($attributes['teachers']);
+		if (isset($attributes['associatedCourses'])) {
+			$course->suppliers()->attach($attributes['associatedCourses']);
 		}
 
-		return view('courses.details', compact('courseKey'));
+		$teachers = $course->teachers->map(function ($user) {
```

Comment:
Este map me parece redundante. Sobre todo porque si ya le vamos a pasar el course, el course ya trae los teachers entonces no sé por qué querríamos pasarlo como una variable adicional, y encima hacer la proyección de los fields no aporta mucho (si no se usan en el view sólo no se usan y ya)

Reply:
Vagamente recuerdo que hice pruebas y no se estaban enviando los teachers dentro de `$course` (?). I'll double check that before deciding what to do about this.